<?xml version="1.0"?>
<launch>
    <rosparam command="load" file="$(find mcr_joint_space_dynamics)/ros/test/only_arm_param_dump.yaml" />
  <arg name="input_topic" default="/joint_states" />
  <arg name="node_name" default="mcr_wrench_estimation" />
  
  <arg name="node_name_joint_dynamics" default="mcr_joint_space_dynamics" />
  
  <arg name="node_name_torque_comparator" default="mcr_torque_comparator" />

  <include file="$(find mcr_joint_space_dynamics)/ros/launch/arm_joint_space_dynamics_nodelet.launch" >
    <arg name="input_topic" value="$(arg input_topic)" />
    <arg name="node_name" value="$(arg node_name_joint_dynamics)" />
    <arg name="manager" value="nodelet_$(arg node_name)" />
    <arg name="type" value="standalone" />
  </include>

  
  <include file="$(find mcr_torque_comparator)/ros/launch/torque_comparator_nodelet.launch" >
    <arg name="input_topic" value="$(arg input_topic)" />
    <arg name="node_name" value="$(arg node_name_torque_comparator)" />
    <arg name="manager" value="nodelet_$(arg node_name)" />
    <arg name="type" value="standalone" />
  </include>

  <include file="$(find mcr_wrench_estimation)/ros/launch/wrench_estimation_nodelet.launch" >
    <arg name="input_topic" value="$(arg input_topic)" />
    <arg name="node_name" value="$(arg node_name)" />
    <arg name="manager" value="nodelet_$(arg node_name)" />
    <arg name="type" value="standalone" />
  </include>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mcr_joint_space_dynamics)/ros/test/test_rviz_config.rviz"/>  
</launch>
